content/content/Advance Usage/custom plugins.md

```diff
@@ -0,0 +1,144 @@
+---
+title: "Custom Plugins"
+date: 2021-11-16T12:38:44-05:00
+draft: false
+---
+
+### How to develop a plugin binary
+
+This document covers how to write a plugin binary using crane-lib. It
+ requires:
+ 1. Go development environment setup
+ 2. Optionally, an overview of the crane toolkit
+ 
+The binary plugin for crane-lib is meant to be a simple Go program in the
+ following format:
+ 
+```
+package main
+
+import (
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/konveyor/crane-lib/transform"
+	"github.com/konveyor/crane-lib/transform/cli"
+)
+
+func main() {
+	fields := []transform.OptionalFields{
+		{
+			FlagName: "MyFlag",
```

Comment:
`FlagName: my-flag` now that optional field names are downcased before passing the map to the plugin.